### ISSUE TITLE
order by not support string column when query more than one database

### ISSUE THREAD

**Issue by xiaoxiong581** [apache/incubator-shardingsphere#4665]
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/incubator-shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response **more than 7 days** and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
4.0.0
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
Sharding-Proxy(docker registry to deploy)
### Expected behavior
more than one database query support order by use string column
### Actual behavior
not support, only support when route one database, cross database support i test is int、datetime etc, if use string column to order by, query sql is always executing, cann't get result or error
#### create table sql
```sql
CREATE TABLE t_cluster(
    clusterId VARCHAR(64) NOT NULL   COMMENT '数据库集群ID' ,
    clusterName VARCHAR(128) NOT NULL   COMMENT '数据库集群名称' ,
    projectId VARCHAR(64) NOT NULL   COMMENT '项目ID' ,
    createTime DATETIME NOT NULL   COMMENT '创建时间' ,
    updateTime DATETIME NOT NULL   COMMENT '更新时间' ,
    PRIMARY KEY (clusterId)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COMMENT = '数据库集群信息表';
ALTER TABLE t_cluster ADD UNIQUE index_cluster_unique_name(clusterName);
```
#### server.yaml
```yaml
authentication:
  users:
    root:
      password: xxxxx
    migu:
      password: xxxxx
      authorizedSchemas: databasemanager
props:
  sql.show: true
```
#### config-databasemanager.yaml
```yaml
schemaName: databasemanager

dataSources:
  ds_0:
    url: jdbc:mysql://192.168.137.102:3306/databasemanager?serverTimezone=UTC&useSSL=false
    username: root
    password: xxxxx
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50
  ds_1:
    url: jdbc:mysql://192.168.137.102:3307/databasemanager?serverTimezone=UTC&useSSL=false
    username: root
    password: xxxx
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50

shardingRule:
  tables:
    t_cluster:
      actualDataNodes: ds_${0..1}.t_cluster
      tableStrategy:
        none:
      databaseStrategy:
        inline:
          shardingColumn: clusterId
          algorithmExpression: ds_${clusterId.hashCode().abs() % 2}
    t_cluster_instance:
      actualDataNodes: ds_${0..1}.t_cluster_instance
      tableStrategy:
        none:
      databaseStrategy:
        inline:
          shardingColumn: clusterId
          algorithmExpression: ds_${clusterId.hashCode().abs() % 2}
  bindingTables:
    - t_cluster,t_cluster_instance
  defaultDatabaseStrategy:
    none:
  defaultTableStrategy:
    none:
```
#### test sql
```sql
select * from t_cluster order by clusterId;
```
#### enviroment
table t_cluster no data
#### expected result
`return no data`
#### actual result
`cann't get result, query is executing, no response`
![image](https://user-images.githubusercontent.com/19368482/76139087-068dda80-6088-11ea-8e35-20010140bbf8.png)
##### sharding-proxy
![image](https://user-images.githubusercontent.com/19368482/76139104-2c1ae400-6088-11ea-8e45-84f053695108.png)
##### mysql
![image](https://user-images.githubusercontent.com/19368482/76139114-3dfc8700-6088-11ea-9753-7608fdd8f76a.png)

**Comment by xiaoxiong581** [comment]
I submit a similar issue a few days ago #4639, but @terrymanu think is invalid.
but i see document in https://shardingsphere.apache.org/document/current/cn/features/sharding/use-norms/sql/, it support order by query in more than one database
![image](https://user-images.githubusercontent.com/19368482/76180146-1af0e500-61f8-11ea-9d57-16753a1396ff.png)

**Comment by wgy8283335** [comment]
I will handle this issue.

**Comment by kimmking** [comment]
@wgy8283335
This issue may be related with #4680, there is something wrong when merged resultset is no data.

**Comment by xiaoxiong581** [comment]
> @wgy8283335
> This issue may be related with #4680, there is something wrong when merged resultset is no data.





> @wgy8283335
> This issue may be related with #4680, there is something wrong when merged resultset is no data.

**Comment by xiaoxiong581** [comment]
> @wgy8283335
> This issue may be related with #4680, there is something wrong when merged resultset is no data.

It also has this issue when i insert data to table
![image](https://user-images.githubusercontent.com/19368482/76414211-d6b93c80-63d1-11ea-9a09-75b158b090c7.png)

**Comment by wgy8283335** [comment]
@xiaoxiong581 
```
schemaName: databasemanager

dataSources:
  ds_0:
    url: jdbc:mysql://192.168.137.102:3306/databasemanager?serverTimezone=UTC&useSSL=false
    username: root
    password: xxxxx
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50
  ds_1:
    url: jdbc:mysql://192.168.137.102:3307/databasemanager?serverTimezone=UTC&useSSL=false
    username: root
    password: xxxx
    connectionTimeoutMilliseconds: 30000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 50

shardingRule:
  tables:
    t_cluster:
      actualDataNodes: ds_${0..1}.t_cluster
      tableStrategy:
        none:
      databaseStrategy:
        inline:
          shardingColumn: clusterId
          algorithmExpression: ds_${clusterId.hashCode().abs() % 2}
    t_cluster_instance:
      actualDataNodes: ds_${0..1}.t_cluster_instance
      tableStrategy:
        none:
      databaseStrategy:
        inline:
          shardingColumn: clusterId
          algorithmExpression: ds_${clusterId.hashCode().abs() % 2}
  bindingTables:
    - t_cluster,t_cluster_instance
  defaultDatabaseStrategy:
    none:
  defaultTableStrategy:
    none:
```

**Comment by wgy8283335** [comment]
I have traced the program and found that exception happened in 'org.apache.shardingsphere.core.execute.sql.execute.result.StreamQueryResult'.
```
    @Override
    public boolean isCaseSensitive(final int columnIndex) throws SQLException {
        return resultSetMetaData.isCaseSensitive(columnIndex);
    }
```
The information of exception is below:
java.sql.SQLException: Streaming result set com.mysql.jdbc.RowDataDynamic@3a1741d6 is still active. 
No statements may be issued when any streaming result sets are open and in use on a given connection. 
Ensure that you have called .close() on any active streaming result sets before attempting more queries.

**Comment by xiaoxiong581** [comment]
> I have traced the program and found that exception happened in 'org.apache.shardingsphere.core.execute.sql.execute.result.StreamQueryResult'.
> 
> ```
>     @Override
>     public boolean isCaseSensitive(final int columnIndex) throws SQLException {
>         return resultSetMetaData.isCaseSensitive(columnIndex);
>     }
> ```
> 
> The information of exception is below:
> java.sql.SQLException: Streaming result set com.mysql.jdbc.RowDataDynamic@3a1741d6 is still active.
> No statements may be issued when any streaming result sets are open and in use on a given connection.
> Ensure that you have called .close() on any active streaming result sets before attempting more queries.

I want to know if you have get expected result when using my config and query sql?

**Comment by wgy8283335** [comment]
@xiaoxiong581 what is the version of java?

**Comment by xiaoxiong581** [comment]
> @xiaoxiong581 what is the version of java?

I use three method to connect Sharding-Proxy Server，like：java code（use jdbctemplate and hikaricp DataSource）、Dbeaver、mysql command in mysql docker, get same result
java code use 1.8.0_211(Oracle)

Sharding-Proxy registry jdk info：
![image](https://user-images.githubusercontent.com/19368482/76603719-726cb900-6548-11ea-9528-8136d4ea54e8.png)

**Comment by wgy8283335** [comment]
This issue is the same as #4460, the bug will be fixed in 5.0.0
